test/unit/com/unifina/controller/api/PermissionApiControllerSpec.groovy

```diff
@@ -248,18 +241,148 @@ class PermissionApiControllerSpec extends ControllerSpecification {
 		response.json.operation == "read"
 	}
 
+	void "user with share permission to resource can delete another user's permission to same resource"() {
+		setup:
+		controller.permissionService = new PermissionService()
```

Comment:
Unit tests should use mocks instead of real collaborators. Why use the real `PermissionService` here instead of the mock, which is already injected to the controller in `setup()`?

Same question for the other test cases